iPetCare/Application/Services/InvitationService.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Collections.Generic;
+using Application.Interfaces;
+using Application.Services.Utilities;
+using Application.Dtos.Invitations;
+using Domain.Models;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Services
+{
+    public class InvitationService : Service, IInvitationService
+    {
+        public InvitationService(IServiceProvider serviceProvider) : base(serviceProvider)
+        {
+        }
+
+        public async Task<ServiceResponse<CreateInvitationDtoResponse>> CreateInvitationAsync(CreateInvitationDtoRequest dto)
+        {
+            if (CurrentlyLoggedUser == null)
+                return new ServiceResponse<CreateInvitationDtoResponse>(HttpStatusCode.Unauthorized);
+
+            if (CurrentlyLoggedUser.Role != Role.Owner && CurrentlyLoggedUser.Role != Role.Vet)
+                return new ServiceResponse<CreateInvitationDtoResponse>(HttpStatusCode.Forbidden);
+
+            var pet = Context.Pets.Find(dto.PetId);
+            if (pet == null)
+                return new ServiceResponse<CreateInvitationDtoResponse>(HttpStatusCode.BadRequest, "Nie znaleziono zwierzaka");
+
+            var didUserRequest = false;
+
+            if (CurrentlyLoggedUser.Role == Role.Owner)
+            {
+                var owners = pet.OwnerPets.Where(x => x.OwnerId == CurrentlyLoggedUser.Owner.Id);
+                if (!owners.Any())
+                    didUserRequest = false;
+                else didUserRequest = true;
+            }
+
+            var existInvitation = await Context.Requests.Where(x => x.PetId == dto.PetId && x.UserId == dto.UserId).ToListAsync();
```

Comment:
Jeśli tego existInvitation używamy wyłącznie by sprawdzić poniżej czy jest any, to zdaje się, że można  można od razu sprawdzić Where().Any()

Reply:
Oczywiście to nie jest błąd :)